Source/App/EbAppMain.c

```diff
@@ -75,6 +75,95 @@ void AssignAppThreadGroup(uint8_t targetSocket) {
 #endif
 }
 
+void* EbAppCreateThread(
+    void *threadFunction(void*),
+    void *threadContext)
+{
+    void* threadHandle = NULL;
+
+#ifdef _WIN32
+    threadHandle = (void*)CreateThread(
+        NULL,                           // default security attributes
+        0,                              // default stack size
+        (LPTHREAD_START_ROUTINE)threadFunction, // function to be tied to the new thread
+        threadContext,                  // context to be tied to the new thread
+        0,                              // thread active when created
+        NULL);                          // new thread ID
+#else
+    pthread_attr_t attr;
+    struct sched_param param = {
+        .sched_priority = 99
+    };
+    pthread_attr_init(&attr);
+    pthread_attr_setschedpolicy(&attr, SCHED_FIFO);
+    pthread_attr_setschedparam(&attr, &param);
+    pthread_attr_setinheritsched(&attr, PTHREAD_EXPLICIT_SCHED);
+
+    threadHandle = (pthread_t*)malloc(sizeof(pthread_t));
+    if (threadHandle != NULL) {
+        int ret = pthread_create(
+            (pthread_t*)threadHandle,     // Thread handle
+            &attr,                        // attributes
+            threadFunction,               // function to be run by new thread
+            threadContext);
+
+        if (ret != 0) {
+            if (ret == EPERM) {
+                pthread_cancel(*((pthread_t*)threadHandle));
+                free(threadHandle);
+                threadHandle = (pthread_t*)malloc(sizeof(pthread_t));
+                if (threadHandle != NULL) {
+                    pthread_create(
+                        (pthread_t*)threadHandle,       // Thread handle
+                        (const pthread_attr_t*)EB_NULL, // attributes
+                        threadFunction,                 // function to be run by new thread
+                        threadContext);
+                }
+            }
+        }
+    }
+    pthread_attr_destroy(&attr);
+#endif // _WIN32
+
+    return threadHandle;
+}
+
+EB_ERRORTYPE EbAppDestroyThread(
+    void* threadHandle)
+{
+    EB_ERRORTYPE error_return = EB_ErrorNone;
+#ifdef _WIN32
+    WaitForSingleObject(threadHandle, INFINITE);
+    error_return = CloseHandle(threadHandle) ? EB_ErrorNone : EB_ErrorDestroyThreadFailed;
+#else
+    pthread_join(*((pthread_t*)threadHandle), NULL);
+    free(threadHandle);
+#endif // _WIN32
+
+    return error_return;
+}
+
+typedef struct EbAppThreadContext_s {
+    uint32_t         numChannels;
+    EbConfig_t     **configs;
+    EbAppContext_t **appCallbacks;
+} EbAppThreadContext_t;
+
+EB_BOOL inputDone = EB_FALSE;
```

Comment:
Maybe put all global variables together at the top of the source code file (near the other global variable:  keepRunning)

Reply:
Please consider multiple encoding channels to update corresponding inputDone.

Reply:
@intelmark , I prefer to keep variables close to where they are used. @Austin-Hu, yes, will address that.